test/network/index.js

```diff
@@ -26,78 +27,95 @@ const NUMBER_OF_TRANSACTIONS = process.env.NUMBER_OF_TRANSACTIONS || 1000;
 // monitor and interact with them as part of the test.
 const NUMBER_OF_MONITORING_CONNECTIONS = TOTAL_PEERS * 2;
 
-describe(`Start a network of ${TOTAL_PEERS} nodes with address "127.0.0.1", WS ports 500[0-9] and HTTP ports 400[0-9] using separate databases`, () => {
-	const wsPorts = [];
-	_.range(TOTAL_PEERS).map(index => {
-		wsPorts.push(5000 + index);
-	});
-	const configurations = setup.config.generateLiskConfigs(TOTAL_PEERS);
-	let testFailedError;
-
-	before(done => {
-		setup.createNetwork(configurations, done);
-	});
+const wsPorts = [];
+_.range(TOTAL_PEERS).map(index => {
+	wsPorts.push(5000 + index);
+});
+const configurations = setup.config.generateLiskConfigs(TOTAL_PEERS);
+let failedToLaunchNetworkError;
 
-	afterEach(function(done) {
-		if (this.currentTest.state === 'failed') {
-			console.warn(`Test failed: ${this.currentTest.title}`);
-			testFailedError = this.currentTest.err;
-		}
-		done();
+const network = new Network(configurations);
+network.launchNetwork({ enableForging: true })
+	.then(() => {
+		initTestSuite();
+		run();
+	})
+	.catch(err => {
+		failedToLaunchNetworkError = err;
+		initTestSuite();
+		run();
 	});
 
-	after(done => {
-		setup.exit(() => {
-			done(testFailedError);
+function initTestSuite() {
+	describe(`Start a network of ${TOTAL_PEERS} nodes with address "127.0.0.1", WS ports 500[0-9] and HTTP ports 400[0-9] using separate databases`, () => {
```

Comment:
I don't think its good practice to have tests in this pattern. 

Why can't you create `describe` at root level and put `network.launchNetwork({ enableForging: true })` in a `before` block?